### ISSUE TITLE
Logits Index out of range

### ISSUE THREAD

**Issue by elewis787** [hayride-dev/hayride#52]
When running multiple prompts through the AI server without clearly memory by a restart, eventually we can not decode the inputs. 

daemon logs: 
```
[2025-07-15T16:40:44Z ERROR hayride_llama] graph_compute: ggml_backend_sched_graph_compute_async failed with error -1
[2025-07-15T16:40:44Z ERROR hayride_llama] llama_decode: failed to decode, ret = -3
[2025-07-15T16:40:44Z ERROR hayride_llama] get_logits_ith: invalid logits id -1, reason: negative index out of range [0, 0)
```

**Comment by elewis787** [comment]
haven't looked deep into this, but I would assume `get_logits_ith: invalid logits id -1, reason: negative index out of range [0, 0)` is the key difference. I can replicate after running long workloads, which suggests a memory leak vs a more general decode error.
